### PR TITLE
[SPARK-49879][CORE] Move `TransportCipherUtil` to a separate file to eliminate Java compilation warnings

### PR DESCRIPTION
**Pull request by LuciferYang**
### What changes were proposed in this pull request?
Run `build/mvn clean install -pl common/network-common`, we can see the following compilation warnings:

```
[WARNING] [Warn] /Users/yangjie01/SourceCode/git/spark-maven/common/network-common/src/main/java/org/apache/spark/network/crypto/CtrTransportCipher.java:73:11:  
auxiliary class TransportCipherUtil in /Users/yangjie01/SourceCode/git/spark-maven/common/network-common/src/main/java/org/apache/spark/network/crypto/TransportCipher.java should not be accessed from outside its own source file
[WARNING] [Warn] /Users/yangjie01/SourceCode/git/spark-maven/common/network-common/src/main/java/org/apache/spark/network/crypto/GcmTransportCipher.java:63:15:
auxiliary class TransportCipherUtil in /Users/yangjie01/SourceCode/git/spark-maven/common/network-common/src/main/java/org/apache/spark/network/crypto/TransportCipher.java should not be accessed from outside its own source file
```

So this pr moves `TransportCipherUtil` to a separate file to eliminate the aforementioned Java compilation warnings.

### Why are the changes needed?
Fix compilation warnings. 

### Does this PR introduce _any_ user-facing change?
No

### How was this patch tested?
- Pass GitHub Actions
- locally run `build/mvn clean install -pl common/network-common`, no longer have the aforementioned compilation warnings.

### Was this patch authored or co-authored using generative AI tooling?
No

### CODE REVIEW

**Review by MaxGekk — APPROVED** [review]

File: common/network-common/src/main/java/org/apache/spark/network/crypto/TransportCipherUtil.java [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+
+import com.google.common.annotations.VisibleForTesting;
+import com.google.crypto.tink.subtle.Hex;
+import com.google.crypto.tink.subtle.Hkdf;
+
+class TransportCipherUtil {
+    /*
+     * This method is used for testing to verify key derivation.
+     */
+    @VisibleForTesting
```
Comment (dongjoon-hyun) [inline common/network-common/src/main/java/org/apache/spark/network/crypto/TransportCipherUtil.java:32]: Shall we use 2-space indentation, @LuciferYang ? Although it follows the original source, we had better change the style according to the Apache Spark community policy.

**Review by dongjoon-hyun — APPROVED** [review]
+1, LGTM (except a minor indentation comment)

File: common/network-common/src/main/java/org/apache/spark/network/crypto/TransportCipherUtil.java (hunk 1 above)
Comment (LuciferYang) [reply common/network-common/src/main/java/org/apache/spark/network/crypto/TransportCipherUtil.java]: [3b10ead](https://github.com/apache/spark/pull/48352/commits/3b10ead94b4f9f2b297ab412241711d12776f8b7) reformatted `TransportCipherUtil.java`

**Comment by dongjoon-hyun** [comment]
Thank you, @LuciferYang .

**Review by dongjoon-hyun — APPROVED** [review]
Pending CIs.

**Comment by dongjoon-hyun** [comment]
Merged to master. Thank you, @LuciferYang and @MaxGekk .

**Comment by LuciferYang** [comment]
Thanks @dongjoon-hyun and @MaxGekk

**Pull request closed without merging**